server/aesdsocket.c

```diff
@@ -195,6 +208,52 @@ static void* client_thread (void* param_v) {
 		max_read -= read_sz;
 	}
 
+#ifdef USE_AESD_CHAR_DEVICE
```

Comment:
@georgehodgkins when there are large changes in logic like this between implementations consider moving into a function rather than blocking out ~100 lines of #ifdef/#endif, this makes your code a bit easier to read.